data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,247 @@
+    if [[ -n $pid ]];then
+        kill -9 $pid >> /dev/null 2>&1;sync
+    else
+        echo -e "PIDs $(basename $0) were killed\n" >> ${monitor_log_file}
+    fi
+
+    echo -e "Kill all associated "virsh console" processes\n" >> ${monitor_log_file}
+
+    while :
+    do
+        #wait for checking all associated "virsh console" processes"
+        sleep 3
+
+        #Check all associated "virsh console" processes
+        ps aux | grep "virsh console" | grep -v "grep" >> /dev/null 2>&1
+        if [[ $? -eq 0 ]];then 
+            sub_pid=`ps aux | grep "expect -c" | grep -v "grep"| awk '{print $2}'`
+            if [[ -n $sub_pid ]];then
+                kill -9 $sub_pid >> /dev/null 2>&1;sync
+            fi
+        else
+            break
+        fi
+    done
+}
+
+function end_monitor() {
+
+    #end of monitor for guest console
+    if [ -f ${MONITOR_LOCKED_FILE} ];then
+        echo -e "End of Monitor for guest console:\n" | tee -a ${monitor_log_file}
+        #UNLOCK Monitor for guest console
+        rm -rf ${MONITOR_LOCKED_FILE} >> /dev/null 2>&1
+        #Kill Pids from Monitor for guest console
+        close_console
+    fi
+}
+
+function start_monitor() {
+
+    #Start monitor for guest console
+    ##After boot up all existed vm guest(s) from set_guest_running func
+    ##Figure out vm_guestnames_state_running_array from all existed and boot up vm guest(s) 
+    get_vm_guestnames_state_running=`virsh list --state-running | grep "${vm_guestnames_types}" | awk '{print $2}'`
+    vm_guestnames_state_running_array=$(echo -e ${get_vm_guestnames_state_running})
+
+    while :
+    do
+        if [ -f ${MONITOR_LOCKED_FILE} ];then
+            #Check with LOCK-UP status
+            cat ${MONITOR_LOCKED_FILE} | grep "LOCKED" >> /dev/null 2>&1
+            if [ $? -eq 0 ];then
+                #KEEP LOCK-UP status
+                #To let just only one virsh console with one vm guest as backgroup PID
+
+                #wait for checking background PID of virsh console
+                sleep 3
+		#Check with background PID of virsh console
+		for vmguest in ${vm_guestnames_array[@]};do
+		    virsh list --state-running | grep ${vmguest}
+                    if [[ $? -eq 0 ]];then
+                        ps aux | grep "virsh console ${vmguest}" | grep -v grep
+                        if [[ $? -ne 0 ]];then
+                            get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+                        fi
+                    else
+                        virsh list --inactive | grep ${vmguest} | grep -v grep
+                        if [[ $? -eq 0 ]];then
+                            echo "${vmguest} is not running now\n" >> ${monitor_log_file}
+                            echo "No any output from virsh console ${vmguest}\n" >> ${monitor_log_file}
+                        fi
+                    fi
+                done
+            else
```

Comment:
This else branch handles the case that the lock file exists but no LOCKED, why handle like below? Print "UNLOCK MONITOR" to log file but does not do the real unlock operation(nothing is done here exept exit). BTW exit makes the script really exit without doing anything further. Why choose such actions for the situation?

Reply:
Good question, this else branch just follow make sure if LOCK-UP status as **LOCKED** mark, if not **LOCKED** mark , used this else branch to print UNLOCK to ${monitor_log_file} and then call exit to terminate this while loop as plan. 

yes, seems this else branch  do not make sense "UNLOCK MONITOR" workflow here. 
will remove this else branch to make sense "UNLOCK MONITOR" workflow as quick updated.

Reply:
Had removed **this else branch** already to make sense "UNLOCK MONITOR" workflow.

For now, just only used with MONITOR_LOCKED_FILE to SETUP LOCKED.

Reply:
DONE